apps/api/v2/src/modules/users/users.repository.ts

```diff
@@ -248,4 +249,46 @@ export class UsersRepository {
       where: { id: userId },
     });
   }
+
+  async getUserAdminTeams(userId?: number) {
+    return await this.dbRead.prisma.user.findFirst({
+      where: {
+        id: userId,
+      },
+      select: {
+        id: true,
+        username: true,
+        teams: {
+          where: {
+            accepted: true,
+            OR: [
+              {
+                role: { in: [MembershipRole.ADMIN, MembershipRole.OWNER] },
+              },
+              {
+                team: {
+                  parent: {
+                    members: {
+                      some: {
+                        id: userId,
+                        role: { in: [MembershipRole.ADMIN, MembershipRole.OWNER] },
+                      },
+                    },
+                  },
+                },
+              },
+            ],
+          },
+          select: {
+            team: {
+              select: {
+                id: true,
+                isOrganization: true,
+              },
+            },
+          },
+        },
+      },
+    });
+  }
```

Comment:
we should already have the different repository functions necessary to check if a user is OrgOwner/OrgOwner/TeamOwner/TeamAdmin, in v2/src/modules/auth/guards/roles/roles.guard.ts